src/bson.rs

```diff
@@ -430,3 +466,121 @@ impl Bson {
         Bson::Document(values)
     }
 }
+
+/// Error for representing failure of getting BSON value
+#[derive(Debug)]
+pub struct GetBsonValueError;
+
+impl fmt::Display for GetBsonValueError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("GetBsonValueError")
+    }
+}
+
+impl Error for GetBsonValueError {
+    fn description(&self) -> &str {
+        "failed to get bson value"
+    }
+}
+
+/// Type alias of `Result<T, GetBsonValueError>`
+pub type GetBsonValueResult<T> = Result<T, GetBsonValueError>;
+
+/// Value helpers
+impl Bson {
+    /// If `Bson` is `FloatingPoint`, return its value. Returns `None` otherwise
+    pub fn as_f64(&self) -> GetBsonValueResult<f64> {
```

Comment:
Hmm. Well, I use `as_*` is because JSON libraries are using this name style.

But yeah, `try_as_f64` is indeed better.